java/src/jmri/InstanceManagerAutoInitialize.java

```diff
@@ -0,0 +1,23 @@
+package jmri;
+
+/**
+ * Provide a hint to the {@link jmri.InstanceManager} that this object needs
+ * have additional initialization performed after it is made available by the
+ * InstanceManager. This allows two classes that have circular dependencies on
+ * being able to get the default instance of each other to be managed
+ * successfully.
+ * <p>
+ * Note: the need to have a class implement this probably is indicative of other
+ * design issues in the implementing class.
+ *
+ * @author Randall Wood Copyright 2017
+ */
+public interface InstanceManagerAutoInitialize {
```

Comment:
For whichever one of those sequences is correct ( and I think the second one is more likely to be correct ) the UML sequence diagram should be added to the code and the javadoc.

We can generate the sequence diagrams on the fly, so the UML description just gets added to the headers. See https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrix/lenz/XNetProgrammer.java for an example.